#50 - CRUD for schools frontend

AmonDeShir opened the pull request:
Should close #11 and close #50 and close #113.

## PC
![image](https://github.com/user-attachments/assets/0cfaa911-e27a-4d3a-b3dd-9e2c85a1363c)
![image](https://github.com/user-attachments/assets/53465c6c-e7de-4fd5-985f-23548a0ed996)
![image](https://github.com/user-attachments/assets/1a3a6dd6-02be-4d59-bdd7-f0805fe9ad51)
![image](https://github.com/user-attachments/assets/d526e167-61a9-4cc7-8413-6f2ffe4d2cb7)
![image](https://github.com/user-attachments/assets/c3af4a81-379c-4611-bfd7-f6c0a059fe33)


## Mobile
![image](https://github.com/user-attachments/assets/ce1281ef-7cd9-4151-9b98-5d2d880a81db)
![image](https://github.com/user-attachments/assets/e6372407-2bab-4d72-8394-a63dbda0fcc6)
![image](https://github.com/user-attachments/assets/eddc42ec-2551-47e9-ad86-99ad21ea5a56)
![image](https://github.com/user-attachments/assets/7b4631ee-89a6-4690-b5fc-062fcd80144e)

JokeUrSelf commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -1,11 +1,126 @@
+    <template #actions>
+      <Expand />
+
+      <ButtonFrame
```
Since you are using `ButtonFrame` as a button, maybe need to get rid of the `Button` component?

AmonDeShir replied:
I changed ButtonFrame to Button. Button sounds more intuitive and immediately makes it clear that this component is a button.

JokeUrSelf commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -1,11 +1,126 @@
+        :class="{'cursor-pointer': status}"
+        @click="startFetching()"
+      >
+        <PlusCircleIcon class="size-6 text-white" /> Importuj
```
Maybe change "Importuj" to "Importuj szkoły", lub "Importuj listę szkół" for the consistency with "Dodaj szkołę"?
Maybe change an `PlusCircleIcon` to `ArrowUpCircleIcon`

AmonDeShir replied:
fixed

JokeUrSelf reviewed (commented):
In quizzes panel, after I click "Dodaj test", this happens:
![image](https://github.com/user-attachments/assets/6275823f-8a27-4f32-a788-fc8c899304ce)

On SchoolsPanel initial load the "importuj" button is disabled, is that ok?
![image](https://github.com/user-attachments/assets/d9110df7-f282-41a8-944e-6f49f3ff9c7a)

When I try to resize a `textarea` with `vDynamicTextAreaHeight` it feels pretty junky, it resets the height I'm trying to set each subsecond. Either disable the ability to resize the `textarea` or make it update only on input changes

AmonDeShir commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -1,11 +1,126 @@
+      <InputWrapper
+        :has-content="!!item.name || editing"
+        :error="errors.name"
+        :show-error="editing"
```
fixed

AmonDeShir commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,101 @@
+<template>
+  <div class="flex flex-col w-full pb-3">
+    <div data-name="toolbar" class="flex xs:flex-row px-4 gap-1 sm:gap-2" :class="{ 'flex-col': mobileNav }">
+      <Dropdown class-btn="rounded-lg" class="mr-auto" :options="options" title="Sortuj">
```
fixed

AmonDeShir commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -1,11 +1,126 @@
+        :has-content="!!item.name || editing"
+        :error="errors.name"
+        :show-error="editing"
+        class="h-full xs:h-auto"
```
fixed

AmonDeShir commented on resources/js/Pages/Admin/Quizzes.vue:
```diff
@@ -1,73 +1,42 @@
+    resource-name="quizzes"
+    new-button-text="Dodaj test"
+    :new-item-data="{ title: 'Nowy test' }"
+    force-full-screen-nav
```
deleted

AmonDeShir commented on resources/js/Helpers/vDynamicTextAreaHeight.ts:
```diff
@@ -0,0 +1,30 @@
+    return
+  }
+
+  input.style.height = '5px'
+  input.style.height = `${input.scrollHeight}px`
```
Resetting height to '5px' is necessary to refresh `scrollHeight`, which ensures that the textarea will shrink when text is removed. Without this reset, `scrollHeight` doesn’t update correctly, causing the textarea to stay larger than needed.

AmonDeShir commented:
> On SchoolsPanel initial load the "importuj" button is disabled, is that ok?

Yes, that's intentional. Only one admin can import schools at a time, so the button remains disabled until the server confirms no other import is in progress.

JokeUrSelf commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -85,7 +84,7 @@ function startFetching() {
           :disabled="!editing"
           placeholder=""
-          class="size-full xs:h-8 bg-transparent outline-none font-bold text-lg xs:mb-0"
+          class="size-full xs:h-8 bg-transparent outline-none font-bold text-lg xs:mb-0 resize-none"
```
I see that you decided to disable the ability to resize the input with `resize-none`. Maybe it is a good idea to add  `input.style.resize = 'none'` line `vDynamicTextAreaHeight.ts` > `initDynamicHeightCalc` instead? That way this style doesn't neet to be manually added every time someone uses `vDynamicTextAreaHeight`.

AmonDeShir replied:
Great idea, implemented.

EwelinaSkrzypacz commented on resources/js/Pages/Admin/SchoolsPanel.vue:
```diff
@@ -1,11 +1,127 @@
+        <CrudInput
+          v-model="data.item.regon"
+          name="regon"
+          label="Regon:"
```
```suggestion
          label="REGON:"
```

EwelinaSkrzypacz reviewed (commented):
1. When I click "Dodaj szkołę" it takes too long to see create school form.
2. Please translate validation message

![image](https://github.com/user-attachments/assets/d3d3ea05-5c3b-4550-b039-fb502b6eabd4)

3. Please add search to school index

4. We should add better validation for REGON and zip code - now I can fill inputs with just "1" and it works, but it shouldn't

5. Please take care of too long inputs

![image](https://github.com/user-attachments/assets/959cf0c6-125c-4610-a4dd-5133e3395607)

6.When we click button "Importuj szkoły" and after importing we will click it one more time, nothing happens. Maybe we should block this button? And on hover show some information that schools are already imported? But what happens, when new school will be added or some school change data, for example address?

AmonDeShir commented:
2. fixed
4. fixed

EwelinaSkrzypacz reviewed (commented):
1. Please work on empty state, because now it looks... sad

![image](https://github.com/user-attachments/assets/3a8b7d87-6c97-4ad1-acf0-e030e4cb7349)

2. When I fill search I have to click on magnifying glass or press Enter - I think this serach should work without clicking - when user enters the text, it should search with some delay

3. I believe it should be "nazwa"

![image](https://github.com/user-attachments/assets/053df255-5651-4833-a3e4-3c272319f9fe)

4. Please fix warnings in console

![image](https://github.com/user-attachments/assets/55e56c09-2d33-474b-8dcb-7d9fe280dcc1)

5. When I click "x" icon nothing happens

EwelinaSkrzypacz commented:
Conflicts :(

AmonDeShir commented:
> 1. Please work on empty state, because now it looks... sad

Fixed.

> 2. When I fill search I have to click on magnifying glass or press Enter - I think this serach should work without clicking - when user enters the text, it should search with some delay

Fixed.

> 3. I believe it should be "nazwa"

To do.

> 4. Please fix warnings in console

Fixed.
> 5. When I click "x" icon nothing happens

Fixed.

EwelinaSkrzypacz commented on resources/js/Pages/Admin/Quizzes.vue:
```diff
@@ -88,15 +75,28 @@ function setQuizzesSorter(type: 'name' | 'creationDate' | 'modificationDate', de
+    </template>
+
+    <template #noContent="{search}">
+      <NoContent :description="search ? `Wygląda na to że nie mamy tego czego szukasz.` : undefined">
```
```suggestion
      <NoContent :description="search ? `Wygląda na to, że nie mamy tego, czego szukasz.` : undefined">
```

EwelinaSkrzypacz commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,148 @@
+    <div v-auto-animate class="flex flex-col gap-4 p-4">
+      <template v-if="newItemMode">
+        <slot name="newItem" :new-item-mode="newItemMode">
+          <CrudNewItem :new-item-data="newItemData" :resource-name="resoisSearchbarEmptyurceName" @done="newItemMode = false">
```
`resource-name` is written correctly?

AmonDeShir replied:
No, it isn't. Fixed

EwelinaSkrzypacz reviewed (commented):
1. When I click this button :heavy_check_mark: (in creating school form) I see 404 error.

2. Please fix warning in console

![image](https://github.com/user-attachments/assets/0e6b8b3e-10ee-4f06-a0a8-5e783efaf6c0)

EwelinaSkrzypacz reviewed (commented):
I can't import schools - I see in console

  `2024-12-02 08:00:32 App\Jobs\FetchSchoolsJob ................. 129.56ms FAIL`


In logs I see error

`[2024-12-02 08:00:32] local.ERROR: cURL error 60: SSL certificate problem: certificate has expired (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://api-rspo.mein.gov.pl/api/placowki?zlikwidowana=0&typ_podmiotu_id=14&wojewodztwo_nazwa=DOLNO%C5%9AL%C4%84SKIE&page=1 {"exception":"[object] (Saloon\\Exceptions\\Request\\FatalRequestException(code: 0): cURL error 60: SSL certificate problem: certificate has expired (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://api-rspo.mein.gov.pl/api/placowki?zlikwidowana=0&typ_podmiotu_id=14&wojewodztwo_nazwa=DOLNO%C5%9AL%C4%84SKIE&page=1 at /application/vendor/saloonphp/saloon/src/Http/Senders/GuzzleSender.php:116)
[stacktrace]`

AmonDeShir commented:
The API URL has changed from api-rspo.mein.gov.pl to api-rspo.men.gov.pl. I've fixed it, everything should work now.

EwelinaSkrzypacz reviewed (commented):
1. Can we hide pagination, when we have 0 results? Also it should be "Brak danych" not "Brak Danych"

![image](https://github.com/user-attachments/assets/36879b16-0993-4d73-bf8b-67a99deab742)

2. Can we update screens on pr desription?

3. Can we rename somehow these two otpions?

![image](https://github.com/user-attachments/assets/aaf2a9da-4bc9-4015-b9aa-2ef7d7227baa)

4. For me, the newest added schools should be first (or maybe recently updated will be better :thinking:) by default

PrabuckiDominik commented on resources/js/Pages/Admin/QuizDemo.vue:
```diff
@@ -1,13 +1,14 @@
-import QuizLayout from '@/components/UserQuiz/QuizLayout.vue'
+import QuizPage from '@/components/UserQuiz/QuizPage.vue'
 import { converter } from '@/Helpers/Converter'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/Helpers/Sorter.ts:
```diff
@@ -0,0 +1,43 @@
+import {computed, type Ref, ref, watch} from 'vue'
+import {router} from '@inertiajs/vue3'
+import {useParams} from '@/Helpers/Params'
```
```suggestion
import { computed, type Ref, ref, watch } from 'vue'
import { router } from '@inertiajs/vue3'
import { useParams } from '@/Helpers/Params'
```

PrabuckiDominik commented on resources/js/Pages/Admin/Quizzes.vue:
```diff
@@ -1,83 +1,70 @@
+import CrudPage from '@/components/Crud/CrudPage.vue'
+import FormButton from '@/components/Common/FormButton.vue'
+import { PlusCircleIcon } from '@heroicons/vue/20/solid'
+import {useParams} from '@/Helpers/Params'
+import NoContent from '@/components/Common/NoContent.vue'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { useParams } from '@/Helpers/Params'
import NoContent from '@/components/Common/NoContent.vue'
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/Pages/Admin/Ranking.vue:
```diff
@@ -3,14 +3,15 @@ import {computed} from 'vue'
 import Divider from '@/components/Common/Divider.vue'
 import FormButton from '@/components/Common/FormButton.vue'
 import {groupBy} from '@/Helpers/GroupBy'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/Pages/User/Dashboard.vue:
```diff
@@ -6,11 +6,12 @@ import FormButton from '@/components/Common/FormButton.vue'
 import Divider from '@/components/Common/Divider.vue'
 import LinkButton from '@/components/Common/LinkButton.vue'
 import QuizItem from '@/components/Dashboard/QuizItem.vue'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/Pages/User/Edit.vue:
```diff
@@ -1,13 +1,14 @@
 <script setup lang="ts">
 import { useForm } from '@inertiajs/vue3'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/Pages/User/Profile.vue:
```diff
@@ -3,11 +3,13 @@ import { type Errors } from '@inertiajs/core'
 import PasswordUpdate from '@/components/Profile/PasswordUpdateSection.vue'
 import UserData from '@/components/Profile/UserDataSection.vue'
 import { Head } from '@inertiajs/vue3'
+import {type PageProps} from '@/Types/PageProps'
```
```suggestion
import { type PageProps } from '@/Types/PageProps'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudItem.vue:
```diff
@@ -0,0 +1,130 @@
+import useRequestResolution from '@/Helpers/RequestResolution'
+import InputWrapper from '@/components/QuizzesPanel/InputWrapper.vue'
+import RequestWrapper from '@/components/Common/RequestWrapper.vue'
+import {ref} from 'vue'
```
```suggestion
import { ref } from 'vue'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudItem.vue:
```diff
@@ -0,0 +1,130 @@
+import { vAutoAnimate } from '@formkit/auto-animate'
+import vDynamicInputWidth from '@/Helpers/vDynamicInputWidth'
+import WarningMessageBox from '@/components/Common/WarningMessageBox.vue'
+import {type Errors, type RequestPayload} from '@inertiajs/core'
```
```suggestion
import { type Errors, type RequestPayload } from '@inertiajs/core'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudNewItem.vue:
```diff
@@ -0,0 +1,62 @@
+import RequestWrapper from '@/components/Common/RequestWrapper.vue'
+import { CheckIcon, XMarkIcon } from '@heroicons/vue/24/outline'
+import { vAutoAnimate } from '@formkit/auto-animate'
+import {type Errors, type RequestPayload} from '@inertiajs/core'
+import {ref} from 'vue'
```
```suggestion
import { type Errors, type RequestPayload } from '@inertiajs/core'
import { ref } from 'vue'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,148 @@
+import { vAutoAnimate } from '@formkit/auto-animate'
+import Expand from '@/components/Common/Expand.vue'
+import Dropdown from '@/components/Common/Dropdown.vue'
+import {type Errors} from '@inertiajs/core'
+import {ref, watch} from 'vue'
```
```suggestion
import { type Errors } from '@inertiajs/core'
import { ref, watch } from 'vue'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,148 @@
+import Button from '@/components/Common/Button.vue'
+import CrudNewItem from '@/components/Crud/CrudNewItem.vue'
+import CrudItem from '@/components/Crud/CrudItem.vue'
+import {useSorter} from '@/Helpers/Sorter'
```
```suggestion
import { useSorter } from '@/Helpers/Sorter'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,148 @@
+import {useSorter} from '@/Helpers/Sorter'
+import SearchBar from '@/components/Crud/SearchBar.vue'
+import Pagination from '@/components/Common/Pagination.vue'
+import {useParams} from '@/Helpers/Params'
```
```suggestion
import { useParams } from '@/Helpers/Params'
```

PrabuckiDominik commented on resources/js/components/Crud/SearchBar.vue:
```diff
@@ -0,0 +1,39 @@
+<script setup lang="ts">
+import {ref} from 'vue'
```
```suggestion
import { ref } from 'vue'
```

JokeUrSelf commented on resources/js/components/Crud/SearchBar.vue:
```diff
@@ -0,0 +1,39 @@
+</script>
+
+<template>
+  <div class="flex gap-1 bg-white/70 text-primary duration-200 p-1 rounded-xl w-full border border-primary/5">
```
```suggestion
  <div class="flex gap-1 bg-white/70 text-primary duration-200 p-1 rounded-xl w-full border border-primary/30">
```

PrabuckiDominik commented on resources/js/Pages/Admin/Quizzes.vue:
```diff
@@ -1,83 +1,70 @@
-import { vAutoAnimate } from '@formkit/auto-animate'
-import { ArrowsUpDownIcon } from '@heroicons/vue/24/outline'
-import { PlusCircleIcon } from '@heroicons/vue/20/solid'
+import {provide, type Ref, ref } from 'vue'
+import {Head} from '@inertiajs/vue3'
```
```suggestion
import { provide, type Ref, ref } from 'vue'
import { Head } from '@inertiajs/vue3'
```

PrabuckiDominik commented on resources/js/Pages/Admin/Ranking.vue:
```diff
@@ -3,14 +3,15 @@ import {computed} from 'vue'
 import Divider from '@/components/Common/Divider.vue'
 import FormButton from '@/components/Common/FormButton.vue'
 import {groupBy} from '@/Helpers/GroupBy'
```
```suggestion
import { groupBy } from '@/Helpers/GroupBy'
```

PrabuckiDominik commented on resources/js/Pages/User/QuizResult.vue:
```diff
@@ -6,8 +6,9 @@ import {computed} from 'vue'
 import LinkButton from '@/components/Common/LinkButton.vue'
 import {Head} from '@inertiajs/vue3'
```
```suggestion
import { Head } from '@inertiajs/vue3'
```

PrabuckiDominik commented on resources/js/components/Crud/CrudPage.vue:
```diff
@@ -0,0 +1,151 @@
+<script setup lang="ts" generic="T extends { id: string | number }, N">
+import {PlusCircleIcon} from '@heroicons/vue/20/solid'
+import {ArrowsUpDownIcon} from '@heroicons/vue/24/outline'
+import {vAutoAnimate} from '@formkit/auto-animate'
```
```suggestion
import { PlusCircleIcon } from '@heroicons/vue/20/solid'
import { ArrowsUpDownIcon } from '@heroicons/vue/24/outline'
import { vAutoAnimate } from '@formkit/auto-animate'
```

PrabuckiDominik reviewed (changes requested):
Please be consistent.
